Src/IronPython.Modules/_collections.cs

```diff
@@ -267,8 +267,9 @@ public deque([ParamDictionary]IDictionary<object, object> dict, params object[]
                 lock (_lockObj) {
                     int found = -1;
                     int startVersion = _version;
-                    WalkDeque(delegate(int index) {
-                        if (PythonOps.EqualRetBool(_data[index], value)) {
+                    WalkDeque((int index) => {
+                        var data = _data[index];
+                        if (ReferenceEquals(data, value) || PythonOps.EqualRetBool(data, value)) {
```

Comment:
Do we want to change PythonOps.EqualRetBool to do the ReferenceEquals check?

Reply:
I believe `PythonOps.EqualRetBool` is meant to be equivalent to `a == b`. For the enumerable stuff what we want is essentially `a is b or a == b`. I don't think we should change `EqualRetBool` since it would change the semantics, but I would have no problem with adding a new method (something like `IsOrEqualRetBool` maybe?)

Reply:
I think that's a good idea